model/task/task.go

```diff
@@ -1988,6 +1988,45 @@ func GetTimeSpent(tasks []Task) (time.Duration, time.Duration) {
 	return timeTaken, latestFinishTime.Sub(earliestStartTime)
 }
 
+// GetTaskResultsByVersion gets all tasks for a specific version
+// Query results can be filtered by task name, variant name and status in addition to being paginated and limited
+func GetTaskResultsByVersion(versionID, taskOrVariantName, sortBy string, statuses []string, sortDir, page, limit int) ([]Task, error) {
+	tasks := []Task{}
+	match := bson.M{
+		VersionKey: versionID,
+	}
+	pipeline := []bson.M{
+		{"$match": &match},
+	}
+	if len(statuses) > 0 {
+		statusFilters := []bson.M{}
+		for _, status := range statuses {
+			statusFilters = append(statusFilters, bson.M{StatusKey: status})
+		}
+		match["$or"] = statusFilters
+	}
+	if len(taskOrVariantName) > 0 {
+		match["$or"] = []bson.M{
+			bson.M{DisplayNameKey: bson.M{"$regex": taskOrVariantName, "$options": "i"}},
+			bson.M{BuildVariantKey: bson.M{"$regex": taskOrVariantName, "$options": "i"}},
+		}
+	}
+	if len(sortBy) > 0 {
+		pipeline = append(pipeline, bson.M{"$sort": bson.D{{Key: sortBy, Value: sortDir}, {Key: "_id", Value: 1}}})
+	}
+	if page > 0 {
+		pipeline = append(pipeline, bson.M{"$skip": page * limit})
+	}
+	if limit > 0 {
+		pipeline = append(pipeline, bson.M{"$limit": limit})
+	}
+	err := db.Aggregate(Collection, pipeline, &tasks)
```

Comment:
There aren't any features of this query that require an aggregation, so let's use a simple query for this. You should get better performance from the query planner since it'll examine the query in its entirety